eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/MutableList.java

```diff
@@ -273,20 +309,35 @@
      * Returns a new MutableList in reverse order.
      */
     @Override
-    MutableList<T> toReversed();
+    default MutableList<T> toReversed()
+    {
+        return this.toList().reverseThis();
```

Comment:
This is also surprising, given the impl of ```toList()```. If I call toReversed() on a list, I'd want the same type back, not one from the factory.

Reply:
While newEmpty had a strange implementation, presumably toList() would be implemented to return the same type, and then this one would overall return the correct type.